janus_core/calculations/geom_opt.py

```diff
@@ -22,11 +25,39 @@
 from janus_core.helpers.utils import none_to_dict, spacegroup
 
 
+def _set_functions(
+    optimizer: Union[Callable, str], filter_func: Optional[Union[Callable, str]] = None
+) -> tuple[Callable, Optional[Callable]]:
+    """
+    Set optimizer and filter functions.
+
+    Parameters
+    ----------
+    optimizer : Union[Callable, str]
+        Optimization function, or name of function from ase.optimize.
+    filter_func : Optional[Union[Callable], str]]
+        ASE filter function, or name of function from ase.filters or ase.constraints.
+        Default is None.
+
+    Returns
+    -------
+    tuple[Callable, Optional[Callable]]
+        Tuple of optimizer function and filter function, if set.
+    """
+    if isinstance(optimizer, str):
+        optimizer = getattr(ase.optimize, optimizer)
```

Comment:
You can return a default from `getattr`, so that may be useful for providing more helpful errors, or `try`-`except` to say what the issue is?

Reply:
I considered a try...except, but when tested the error it outputs `AttributeError: module 'ase.optimize' has no attribute 'test'`, which combined with the docstrings feels quite descriptive.

Anything you would add specifically?

Reply:
It's very much a philosophical thing, but to my eye, the user shouldn't know or care when running Janus that they're running a `ase.optimize` optimizer. 

I think we should have something like `raise AttributeError(`f"No such optimizer ({optimizer})") from err`, the reason being that if we add other optimizers somewhere down the line, the error is always consistent internally and can be searched for and controlled by us.

Reply:
I suppose my argument would be that at the moment, they have to care, since there are no current alternatives, but I see your point about future proofing, so I'll add something